### PR TITLE
fix(date-picker):fix format type processing for showTime.

### LINKED ISSUE #44275: datepicker设置showtime和多格式匹配时，手工输入时间会报错
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/embed/ri-qi-ge-shi-antd-5-8-3-forked-7m5gl9?fontsize=14&hidenavigation=1&theme=dark)

### Steps to reproduce

手工输入时间，如2023-02-28 00:00

### What is expected?

正常接受定义的多格式输入

### What is actually happening?

报错：TypeError
(formatStr || "").replace is not a function

| Environment | Info |
| --- | --- |
| antd | undefined |
| React | 5.8.3 |
| System | mac |
| Browser | edge |

### PR DESCRIPTION
**Pull request by Zian502**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
fix #44275

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    fix(date-picker):fix showTime work error when format is Array      |
| 🇨🇳 Chinese |    修复 date-picker 组件，当 showTime 为 true 且 format 为数组时，组件报错     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at e4124ab</samp>

Fix date picker time format bug when `format` prop is an array. Use the first element of the array as the time picker format in `components/date-picker/util.ts`.

### <samp>🤖 Generated by Copilot at e4124ab</samp>

* Fix date picker time format bug when `format` prop is an array ([link](https://github.com/ant-design/ant-design/pull/44306/files?diff=unified&w=0#diff-64d79cd59dfc100bce2821bc07efac6651fb728fd8473b730597a06dd0815dceR129-R133))

### CODE REVIEW

**Comment by Zian502** [comment]
https://github.com/ant-design/ant-design/issues/44275

File: components/date-picker/util.ts [hunk 1]
```diff
@@ -126,6 +126,11 @@ export function getTimeProps<DateType, DisabledTime>(
   const firstFormat = toArray(format)[0];
   const showTimeObj = { ...props };
 
+  // https://github.com/ant-design/ant-design/issues/44275
+  if (format && Array.isArray(format)) {
+    showTimeObj.format = firstFormat;
+  }
```
Comment (afc163) [inline components/date-picker/util.ts:132]: 来个测试用例如何

Comment (Zian502) [reply components/date-picker/util.ts:132]: 好的，待会加上

Comment (Zian502) [reply components/date-picker/util.ts:132]: > 来个测试用例如何

done

**Review by AmanKumarVerma11 — APPROVED** [review]

**Review by afc163 — APPROVED** [review]

**Pull request merged**